kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -271,25 +271,67 @@ function showExamReportDetailPage(store, classId, examId) {
   store.dispatch('CORE_SET_PAGE_LOADING', false);
 }
 
-function activateExam() {
-  this.displayModal(false);
+function activateExam(store, examId) {
+  ExamResource.getModel(examId).save({ active: true }).then(
+    () => {
+      const exams = store.state.pageState.exams;
+      const examIndex = exams.findIndex(exam => exam.id === examId);
+      exams[examIndex].active = true;
+
+      store.dispatch('SET_EXAMS', exams);
```

Comment:
Yeah, my questions was whether it is more idiomatic to have the heavy lifting done in the mutation. For example, the vuex todomvc example tracks more with my experience in redux https://github.com/vuejs/vuex/blob/master/examples/todomvc/vuex/store.js

Basically all the action would be doing is dispatching something like 'RENAME_EXAM', plus whatever params are needed to it (e.g. examId, newTitle).

I don't think it really matters, but the rationale in redux-land was that it is much simpler to write unit tests for the different pieces.